tests/test_general.py

```diff
@@ -68,6 +69,68 @@ def test_tilt_desired(self):
         ].empty
 
 
+class TestRTLHeight:
+    # The return to home height changes with the distance from home
+    # mode was triggered
+    # check the height above ground while the drone returns to home. compare it with 
+    # the allowed maximum or minimum heights, until the drone has reached home and motors have been turned off
+
+   def setup_class(self):
+        # get the required data
+        filepath = getfilepath()
+        topics = [
+            "vehicle_local_position",
+            "vehicle_status",
+        ]
+        self.ulog = pyulog.ULog(filepath, topics)
+        self.df = ulogconv.merge(ulogconv.createPandaDict(self.ulog))
+
+
+   def test_rtl(self):
+       # TODO adjust this test to also work with cone (only for yuneec)
+        NAVIGATION_STATE_AUTO_RTL = 5           # see https://github.com/PX4/Firmware/blob/master/msg/vehicle_status.msg
+        max_height_within_RTL_MIN_DIST = 10     # within this radius, the drone should not go higher than this value after RTL has been triggered
+        thresh = 1                              # Threshold for position inaccuracies
+
+        # drone parameters: below rtl_min_dist, the drone follows different rules than outside of it.
+        rtl_min_dist = (
+            loginfo.get_param(self.ulog, "RTL_MIN_DIST", 0)
+        )
+        rtl_return_alt = (
+            loginfo.get_param(self.ulog, "RTL_RETURN_ALT", 0)
+        )
+
+        posanl.add_horizontal_distance(self.df)
+
+        # Check vehicle status for RTL trigger
+        # store all distance and height values that are reached ONLY DURING RTL in a new array
```

Comment:
To store these values in a new array is one option. However, since we are working with pandas dataframe, often there is a faster method that would not require to loop through the entire time-series